src/db/mod.rs

```diff
@@ -16,7 +17,58 @@ mod add_package;
 pub mod blacklist;
 pub mod delete;
 pub(crate) mod file;
-mod migrate;
 mod overrides;
 mod pool;
 pub(crate) mod types;
+
+static MIGRATOR: Migrator = sqlx::migrate!();
+
+pub async fn migrate(conn: &mut sqlx::PgConnection, target: Option<i64>) -> Result<()> {
+    conn.ensure_migrations_table().await?;
+
+    // `database_versions` is the table that tracked the old `schemamama` migrations.
+    // If we find the table, and it contains records, we insert a fake record
+    // into the `_sqlx_migrations` table so the big initial migration isn't executed.
+    if sqlx::query(
+        "SELECT table_name
+         FROM information_schema.tables
+         WHERE table_schema = 'public' AND table_name = 'database_versions'",
+    )
+    .fetch_optional(&mut *conn)
+    .await?
+    .is_some()
+    {
+        let count: i64 = sqlx::query_scalar("SELECT count(*) FROM database_versions")
```

Comment:
Rather than just checking the count, it may be better to check that `SELECT max(version) == 39` (and probably error in the case that the table exists but the version number _isn't_ that).

Reply:
valid point, done